definitions/tools/grolar.cwl

```diff
@@ -0,0 +1,185 @@
+#!/usr/bin/env cwl-runner
+
+cwlVersion: v1.0
+class: CommandLineTool
+label: "grolar- pizzly gene fusion output parser"
+baseCommand: ["Rscript", "grolar.R"]
+requirements:
+    - class: InlineJavascriptRequirement
+    - class: ResourceRequirement
+      coresMin: 1
+      ramMin: 16000
+    - class: DockerRequirement
+      dockerPull: "mgibio/grolar-cwl"
+    - class: InitialWorkDirRequirement
+      listing:
+      - entryname: 'grolar.R'
+        entry: |
+           args <- commandArgs(trailingOnly = TRUE)
+           #adapted from https://github.com/MattBashton/grolar
+           
+           # Load JSON output from pizzly add in gene co-ordinates + compute distance when on same chr
+           
+           # Set up
+           # Change to your own output location
+           
+           # CRAN libs
+           library(jsonlite)
+           library(dplyr)
+           
+           # Assuming you have used GRCh38 gene models
```

Comment:
Adding the grolar step makes the whole rnaseq.cwl pipeline only compatible with GRCh38 references? If so shouldn't we make a note of this elsewhere? Or maybe some sort of check?

Reply:
Good point - that gets tricky, since you need to have the appropriate R package installed for each Ensembl version (and need to specify that as an input!).

One proposed solution:
1) this script should take additional arguments (ensembl version and species) from the pipeline
2) add some logic to convert that to the right package name 
3) It should check to see if that package is installed.  If not, it should install it in a user-configurable directory. 

So we would make `r_library_path` an input, and for our purposes, we'll set it to default to a path in `/core/` to use for these annotations.  The R script can then check whether this path exists at runtime.  If it does, it adds it to the library path. and proceeds to the check/install/run phase.  If the path does not exist, it can toss a warning and install the needed package to `/tmp/`. (which means that it will need to be installed anew every time that the tool is run).  Thoughts?